### PR TITLE
14.0 x2m no open las

### PR DESCRIPTION
**Pull request by smetl**
**[IMP] web: Manage 'no_open' options for x2many fields**

options="{'no_open': '1'}" is managed for many2one fields.
This commit aims to do the same for x2many fields.

**[FIX] account_edi: Prevent opening the EDI documents in any case**

Also, remove the duplicated 'edi_document_ids' field inside the view.
'edi_document_ids' is now in debug mode for payments.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/fields/relational_fields.js [hunk 1]
```diff
@@ -1920,6 +1920,10 @@ var FieldOne2Many = FieldX2Many.extend({
             this.recordParams,
             { additionalContext: params.context }
         ));
+
+        if(this.attrs.options && this.attrs.options.no_open)
+            return;
```
Comment (aab-odoo) [inline addons/web/static/src/js/fields/relational_fields.js:1925]:
```suggestion
        if(this.attrs.options && this.attrs.options.no_open) {
            return;
        }
```
Also, there should be a test, and the option should be documented

File: addons/web/static/src/js/fields/relational_fields.js [hunk 2]
```diff
@@ -1920,6 +1923,11 @@ var FieldOne2Many = FieldX2Many.extend({
             this.recordParams,
             { additionalContext: params.context }
         ));
+
+        if(this.attrs.options && this.attrs.options.no_open) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/fields/relational_fields.js:1927]:
```js
if (this.nodeOptions.no_open) {
```

**Review by aab-odoo — APPROVED** [review]
The condition can be simplified (see comment on the diff). This must be tested and documented, but I agree to merge it as is for now, and write the test and doc afterwards.

**Review by FlorianGilbert — APPROVED** [review]

**Comment by FlorianGilbert** [comment]
@robodoo r+

**Comment by smetl** [comment]
@robodoo rebase-ff

**Comment by FlorianGilbert** [comment]
@robodoo rebase-ff

**Pull request closed without merging**